pkg/controller/job/job_controller_test.go

```diff
@@ -560,6 +560,44 @@ func TestControllerSyncJob(t *testing.T) {
 			expectedCreatedIndexes: sets.NewInt(3, 4),
 			indexedJobEnabled:      true,
 		},
+		"indexed job with running pod adoption from job with larger index range": {
+			parallelism:    2,
+			completions:    5,
+			backoffLimit:   6,
+			completionMode: batch.IndexedCompletion,
+			podsWithIndexes: []indexPhase{
+				{"0", v1.PodSucceeded},
+				{"6", v1.PodRunning},
+				{"7", v1.PodRunning},
+			},
+			jobKeyForget:           true,
+			expectedCreations:      2,
+			expectedSucceeded:      1,
+			expectedDeletions:      2,
+			expectedCompletedIdxs:  "0",
+			expectedCreatedIndexes: sets.NewInt(1, 2),
+			expectedActive:         2,
+			indexedJobEnabled:      true,
+		},
+		"indexed job with succeeded pod adoption from job with larger index range": {
+			parallelism:    2,
+			completions:    5,
+			backoffLimit:   6,
+			completionMode: batch.IndexedCompletion,
+			podsWithIndexes: []indexPhase{
+				{"0", v1.PodSucceeded},
+				{"6", v1.PodSucceeded},
+				{"7", v1.PodSucceeded},
```

Comment:
what about Failed pods with indexes outside the range? They shouldn't count as failures.